src/GraphQL.Tests/Subscription/ObservableExtensionsTests.cs

```diff
@@ -0,0 +1,317 @@
+#nullable enable
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+
+using GraphQL.Subscription;
+
+namespace GraphQL.Tests.Subscription;
+
+public class ObservableExtensionsTests
+{
+    private SampleObservable<string> Source { get; } = new();
+    private SampleObserver Observer { get; } = new();
+
```

Comment:
Let's add such test
```csharp
[Fact]
    public async Task CancelAndClearQueueBeforeSendToObserver()
    {
        var observable = Source
            .SelectCatchAsync(
                async (data, token) =>
                {
                    var s = int.Parse(data);
                    await Task.Delay(s);
                    return data;
                },
                async (error, token) =>
                {
                    error.Message.ShouldBe("abc");
                    return new ApplicationException();
                });
        var subscription = observable.Subscribe(Observer);
        for (int i = 0; i < 1000; ++i)
            Source.Next("2000");
        subscription.Dispose();
        await Observer.WaitForAsync("");
    }
```

Reply:
I do not quite understand what is being tested.

The code as written:
1. Starts 1000 asynchronous transformations simultaneously executing, each one completing after a 2-sec delay (ignoring the cancellation token).  Note: all 1000 transformations will finish in 2 sec, not in 2000 sec.
2. Signals the cancellation token
3. Verifies that no data has as of yet been received by the observer (always true, since they all take 2-sec to execute).
4. Passes the test immediately, not waiting for any of the 1000 transformations to complete (regardless of any logic to clear the queue, monitor the cancellation, or anything else).

I'm going to assume you're trying to verify that pending events do not send.  Adding 1000 events does not seem to serve a purpose.  Either one will send when it is not supposed to, or it won't.  I'm rewriting the test as follows:

```cs
    [Fact]
    public async Task CanceledSubscriptionsDontSendPendingData()
    {
        var observable = Source
            .SelectCatchAsync(
                async (data, token) =>
                {
                    var s = int.Parse(data);
                    await Task.Delay(s);
                    return data;
                },
                (error, token) => new ValueTask<Exception>(error));
        var subscription = observable.Subscribe(Observer);
        Source.Next("200"); // if the value is 0, it completes synchronously, and the test would fail
        Source.Error(new ExecutionError("test")); // a completed synchronous transformation, but in the queue after one with a delay
        subscription.Dispose();
        Observer.Current.ShouldBe("");
        await Task.Delay(1000);
        Observer.Current.ShouldBe("");
    }
```

Reply:
Test added as described above.

Reply:
OK